added instructions for fixing permissions issue"

nicvagn opened the pull request:
- Fix formatting
- Move SDK and our stub application into src/
- Do not link hidapi_hidraw when on macOS, because the former is Linux-only
- Refactor
- Refactor
- Move SDK source files into sdk/ folder
- Generate compile_commands.json
- Set CMAKE_CONFIGURATION_TYPES to Debug and Release
- Refactor to fix clang-tidy warnings
- Add .clang-format and .clang-tidy
- Add .env and justfile
- Add coverage.sh
- Add doxygen setup
- Add .gitignore
- Fixes to change so file name
- Docs
- Use recommended lengths for returned strings
- Improve error handling
- Sort targets alphabetically, simplify targets by removing `cd {{project_dir}}`
- Removing test-related targets because there are no tests yet
- Improve error message when failing to connect to board
- Simplify string printing, rename version to sdk_version
- Re-add better string printing
- Fix typo in .gitignore to ignore .cache/
- Add build instructions
- Improve README
- Support CXX environment variable
- Update README with build instructions, notably for Windows
- Update README
- Add .vs/ for Visual Studio to .gitignore
- Improve Windows build instructions
- Docs
- Use #define to set recommended version lengths
- Add LED example including documentation
- Markdown formatting
- Fix typo
- docs
- docs
- Fix typo
- docs
- Add void to parameter list
- Add example on how to connect to the chess board
- Docs
- Auto-reformat
- Improve API documentation (using doxygen format) plus some minor refactoring (parameter names)
- Refactoring: use size_t instead of int for version functions and cl_get_file()
- Add header guard EASY_LINK_SDK_HEADER_GUARD
- Refactor LED example
- docs
- Set project title
- Configure INPUT settings for doxygen
- Fix code block formatting
- Improve docs
- Configure INPUT settings
- Build docs
- docs
- cl_led(): rename parameter `data` to `leds`
- docs
- Update build.yml
- Add workflow status badge
- Remove -S option
- Show versions of clang and cmake
- Update workflow
- Use cmake 3.20.0+
- Update configure step
- Add GITHUB_TOKEN to environment for checkout step
- Update workflow
- Remove step to show cmake help
- Trigger only on branch main and on PRs for branch main
- Install ninja
- Print ninja version after installation
- Set build type to Release
- Fix path to *.dll and *.lib
- Capitalize build steps
- Use env variable of BUILD_TYPE for path to dll/lib
- Use upload-artifact v3.1.3
- Minor refactoring
- Rename artifact to easylink-windows-x64
- Add note that this repo is a fork
- docs
- Docs
- docs
- Rename "chess board" to "chessboard"
- Fix real-time examples
- Improve battery level example
- Use %d instead of %i because the return value is a signed int
- Improve LED example
- docs
- Docs
- docs
- docs
- Docs
- remove trailing whitespace
- Improve example code
- refactoring
- Use enum instead of macros
- Improve examples
- Remove cl_get_file() call so we don't accidentally delete game files
- Improve docs for cl_get_file()
- docs
- docs
- Add warning for using cl_get_file()
- Update README.md
- Add cl_get_file_and_keep() and cl_get_file_and_delete()
- Update README.md to use cl_get_file_and_delete()
- Update README.md
- Improve example code for versions
- Clarify we skip some error handling for the sake of brevity
- Document frequency range of cl_beep()
- Add error handling to beep example
- Clarify that return values are 1 (true) on success
- Improve error handling for examples, rename led[] to leds[]
- Improve real-time callback example
- Improve real-time callback code
- Improve .clang-format
- Use clang-tidy suggested enum naming
- Clarify cl_get_file() related functions
- Improve docs
- Correct the docs for building
- Improve Windows instructions
- docs
- build docs
- docs
- docs
- Improve error handling
- docs
- Add missing include
- Remove duplicated build section
- docs
- Update README.md
- Write errors to STDERR
- Docs
- Docs
- just: simplify 'run' target
- Add just info
- Add comments, remove duplicate cl_disconnect()
- Remove note about cl_disconnect()
- docs
- docs
- Clarify behavior of game file retrieval of failure conditions
- docs
- docs
- docs
- added fix_permissions_linux, a text file

miguno committed:
Fix formatting

miguno committed:
Move SDK and our stub application into src/

miguno committed:
Do not link hidapi_hidraw when on macOS, because the former is Linux-only

miguno committed:
Refactor

miguno committed:
Refactor

miguno committed:
Move SDK source files into sdk/ folder

miguno committed:
Generate compile_commands.json

miguno committed:
Set CMAKE_CONFIGURATION_TYPES to Debug and Release

miguno committed:
Refactor to fix clang-tidy warnings

miguno committed:
Add .clang-format and .clang-tidy

miguno committed:
Add .env and justfile

miguno committed:
Add coverage.sh

miguno committed:
Add doxygen setup

miguno committed:
Add .gitignore

miguno committed:
Merge remote-tracking branch 'upstream/main'

miguno committed:
Fixes to change so file name

miguno committed:
Docs

miguno committed:
Use recommended lengths for returned strings

miguno committed:
Improve error handling

miguno committed:
Sort targets alphabetically, simplify targets by removing `cd {{project_dir}}`

miguno committed:
Removing test-related targets because there are no tests yet

miguno committed:
Improve error message when failing to connect to board

miguno committed:
Simplify string printing, rename version to sdk_version

miguno committed:
Re-add better string printing

miguno committed:
Fix typo in .gitignore to ignore .cache/

miguno committed:
Add build instructions

miguno committed:
Improve README

miguno committed:
Support CXX environment variable

miguno committed:
Update README with build instructions, notably for Windows

miguno committed:
Update README

miguno committed:
Add .vs/ for Visual Studio to .gitignore

miguno committed:
Improve Windows build instructions

miguno committed:
Docs

miguno committed:
Use #define to set recommended version lengths

miguno committed:
Add LED example including documentation

miguno committed:
Markdown formatting

miguno committed:
Fix typo

miguno committed:
docs

miguno committed:
docs

miguno committed:
Fix typo

miguno committed:
docs

miguno committed:
Add void to parameter list

miguno committed:
Add example on how to connect to the chess board

miguno committed:
Docs

miguno committed:
Auto-reformat

miguno committed:
Improve API documentation (using doxygen format) plus some minor refactoring (parameter names)

miguno committed:
Refactoring: use size_t instead of int for version functions and cl_get_file()

miguno committed:
Add header guard EASY_LINK_SDK_HEADER_GUARD

miguno committed:
Refactor LED example

miguno committed:
docs

miguno committed:
Set project title

miguno committed:
Configure INPUT settings for doxygen

miguno committed:
Fix code block formatting

miguno committed:
Improve docs

miguno committed:
Configure INPUT settings

miguno committed:
Build docs

miguno committed:
docs

miguno committed:
cl_led(): rename parameter `data` to `leds`

miguno committed:
docs

miguno committed:
Update build.yml

miguno committed:
Add workflow status badge

miguno committed:
Remove -S option

miguno committed:
Show versions of clang and cmake

miguno committed:
Update workflow

miguno committed:
Use cmake 3.20.0+

miguno committed:
Update configure step

miguno committed:
Add GITHUB_TOKEN to environment for checkout step

miguno committed:
Update workflow

miguno committed:
Remove step to show cmake help

miguno committed:
Trigger only on branch main and on PRs for branch main

miguno committed:
Install ninja

miguno committed:
Print ninja version after installation

miguno committed:
Set build type to Release

miguno committed:
Fix path to *.dll and *.lib

miguno committed:
Capitalize build steps

miguno committed:
Use env variable of BUILD_TYPE for path to dll/lib

miguno committed:
Use upload-artifact v3.1.3

miguno committed:
Minor refactoring

miguno committed:
Rename artifact to easylink-windows-x64

miguno committed:
Add note that this repo is a fork

miguno committed:
docs

miguno committed:
Docs

miguno committed:
docs

miguno committed:
Rename "chess board" to "chessboard"

miguno committed:
Fix real-time examples

miguno committed:
Improve battery level example

miguno committed:
Use %d instead of %i because the return value is a signed int

miguno committed:
Improve LED example

miguno committed:
docs

miguno committed:
Docs

miguno committed:
docs

miguno committed:
docs

miguno committed:
Docs

miguno committed:
remove trailing whitespace

miguno committed:
Improve example code

miguno committed:
refactoring

miguno committed:
Use enum instead of macros

miguno committed:
Improve examples

miguno committed:
Remove cl_get_file() call so we don't accidentally delete game files

Prevent people who are trying out the SDK from accidentally removing
their game files from internal storage.

miguno committed:
Improve docs for cl_get_file()

miguno committed:
docs

miguno committed:
docs

miguno committed:
Add warning for using cl_get_file()

miguno committed:
Update README.md

miguno committed:
Add cl_get_file_and_keep() and cl_get_file_and_delete()

miguno committed:
Update README.md to use cl_get_file_and_delete()

miguno committed:
Update README.md

miguno committed:
Improve example code for versions

miguno committed:
Clarify we skip some error handling for the sake of brevity

miguno committed:
Document frequency range of cl_beep()

miguno committed:
Add error handling to beep example

miguno committed:
Clarify that return values are 1 (true) on success

miguno committed:
Improve error handling for examples, rename led[] to leds[]

miguno committed:
Improve real-time callback example

miguno committed:
Improve real-time callback code

miguno committed:
Improve .clang-format

miguno committed:
Use clang-tidy suggested enum naming

miguno committed:
Clarify cl_get_file() related functions

miguno committed:
Improve docs

miguno committed:
Correct the docs for building

miguno committed:
Improve Windows instructions

miguno committed:
docs

miguno committed:
build docs

miguno committed:
docs

miguno committed:
docs

miguno committed:
Improve error handling

miguno committed:
docs

miguno committed:
Add missing include

miguno committed:
Remove duplicated build section

miguno committed:
docs

miguno committed:
Update README.md

miguno committed:
Write errors to STDERR

miguno committed:
Docs

miguno committed:
Docs

miguno committed:
just: simplify 'run' target

miguno committed:
Add just info

miguno committed:
Add comments, remove duplicate cl_disconnect()

miguno committed:
Remove note about cl_disconnect()

miguno committed:
docs

miguno committed:
docs

miguno committed:
Clarify behavior of game file retrieval of failure conditions

miguno committed:
docs

miguno committed:
docs

miguno committed:
docs

nicvagn committed:
added fix_permissions_linux, a text file

nicvagn committed:
Update README.md

chessnutech commented:
I'm very sorry for the late reply. I was focusing on new products recently, so I didn't notice it.

Thank you very much for your submission, which makes the project easier to use.

I am happy to accept these changes, but there is a conflicting file to deal with.

I will accept it as soon as the conflict is resolved.

nicvagn committed:
Fixed up so merges cleanly.

Merge branch 'main' of https://github.com/chessnutech/EasyLinkSDK

This was just a quick and dirty merge, but nothing important was
changed.

nicvagn committed:
Merge remote-tracking branch 'origin/main'

nicvagn commented:
I was pretty fast and dirty, but that should maybe work. 

The workflow was done by another developer. I know nothing about workflows.

zx950921 reviewed (commented):
Thank you very much for your support
